src/utilities/storeTransaction.ts

```diff
@@ -0,0 +1,30 @@
+import type { Types } from "mongoose";
+import * as fs from "fs";
+import { TRANSACTION_LOG_PATH } from "../constants";
+
+/**
+ * store log for the database transaction
+ * @remarks
+ * This is a utility method.
+ * @param createdBy - user responisible for creating the transaction
+ * @param type - type of transaction
+ * @param modelName - model name in the database which was changed
+ * @message - transaction message
+ */
+
+export const storeTransaction = (
+  createdBy: string | Types.ObjectId,
+  type: string,
+  modelName: string,
+  message: string
+): void => {
+  const logPath = TRANSACTION_LOG_PATH as string;
+
+  const timestamp = new Date().toISOString();
+  const logMessage = `timestamp=${timestamp}, createdBy=${createdBy}, type=${type}, model=${modelName}, message=${message}\n`;
```

Comment:
1. What does the paid version of mongdb log? We should be logging at least that data. 
2. We only have the fact that someone made an update. We don't have any context related to the values updated or deleted. This means a forensic audit is not possible in the case of fraudulent activity. This is insufficient

Reply:
mongodb logs is actually much detailed, it logs in json format, and example log is something like
`
{
  "timestamp": "2024-01-06T12:34:56.789Z",
  "operation": "update",
  "database": "marketingDB",
  "collection": "advertisements",
  "user": {
    "username": "adminUser",
    "roles": ["dbAdmin"]
  },
  "client": {
    "ip": "192.168.1.100",
    "application": "MongoDB Compass"
  },
  "operationDetails": {
    "query": {"_id": "ad123"},
    "update": {"$set": {"image": "new_image_url.jpg"}},
    "documentId": "ad123"
  },
  "outcome": "success"
}
`
2. I think, for our case, we should atleast add the `update` field, to know what was set/push/pulled from the document. Maybe the client ip too, if possible. What do you think?